src/v/storage/disk_log_impl.cc

```diff
@@ -1731,6 +1734,244 @@ disk_log_impl::make_cached_reader(log_reader_config config) {
       .then([this](auto rdr) { return _readers_cache->put(std::move(rdr)); });
 }
 
+namespace details {
+// This accumulator is used to compute size of the on-disk representation of the
+// record batches
+struct batch_size_accumulator {
+    ss::future<ss::stop_iteration> operator()(model::record_batch b) {
+        vassert(
+          result_size_bytes != nullptr,
```

Comment:
it seems to me like you could add a constructor to this struct and not pay this cost on each operator() invocation?